Lib/xml/etree/ElementTree.py

```diff
@@ -1261,18 +1261,35 @@ def iterator(source):
     gen = iterator(source)
     class IterParseIterator(collections.abc.Iterator):
         __next__ = gen.__next__
+
         def close(self):
             if close_source:
                 source.close()
             gen.close()
+            self._closed = True
```

Comment:
You can simply set `close_source = False`. But it should be declared `nonlocal`.